framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -894,7 +892,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
```

Comment:
An `assert supportedQuals.class == UnmodifiableSet.class` should have been there immediately after assigning to supportedQuals, which would have enforced this contract. Oh well!